### ISSUE TITLE
Reports cannot be write on the shared volume

### ISSUE THREAD

**Issue by omnisnash** [atlassian-labs/connect-security-req-tester#14]
Hey ! Nice project, could be useful to validate security requirement before releasing. But, I have an issue with the Docker usage

# Description

When following the README instruction to run the project in a Docker container, the analyse is done but report(s) cannot be write in the /app/out directory due to a permission issue:

``PermissionError: [Errno 13] Permission denied: 'out'``

I've tested first on a Windows 10 environment using Docker with WSL2 engine. I've observed a weird behaviour: when running the image, a directory named ``'out;C'`` is created, and if I replace the run command with ``docker run -v $(pwd)/'out;C':/app/out atlas-connect-sec-test <url of descriptor>`` then it worked.

I've also tested on Ubuntu 20.4, same permission error, and again a  ``'out;C'` is created

![image](https://user-images.githubusercontent.com/36193051/100260780-1fa62680-2f4a-11eb-90dd-e1e43256bb4e.png)

# Example(s)

````
docker run -v $(pwd)/out:/app/out atlas-connect-sec-test https://elements.valiantys.com/elements/atlassian-connect.json
2020-11-25 17:08:20,097 INFO - tls_scan.py:99 - Starting SSL/TLS Scan for elements.valiantys.com...
2020-11-25 17:08:30,687 INFO - tls_scan.py:115 - SSL/TLS Scan completed.
2020-11-25 17:08:30,688 INFO - descriptor_scan.py:138 - Scanning app descriptor at: https://elements.valiantys.com/elements/atlassian-connect.json
2020-11-25 17:08:46,441 INFO - descriptor_scan.py:162 - Descriptor scan complete, found and visited 10 links
2020-11-25 17:08:46,477 INFO - main.py:40 - Starting analysis of results...
2020-11-25 17:08:46,479 INFO - main.py:52 - Finished analysis
Traceback (most recent call last):
  File "main.py", line 72, in <module>
    fire.Fire(main)
  File "/home/app/.local/lib/python3.8/site-packages/fire/core.py", line 138, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/app/.local/lib/python3.8/site-packages/fire/core.py", line 463, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/app/.local/lib/python3.8/site-packages/fire/core.py", line 672, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "main.py", line 56, in main
    generator.save_report()
  File "/app/reports/generator.py", line 108, in save_report
    self._write_output(final_report, html_name)
  File "/app/reports/generator.py", line 45, in _write_output
    Path(self.out_dir).mkdir(exist_ok=True, parents=True)
  File "/usr/local/lib/python3.8/pathlib.py", line 1287, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: 'out'
````

# Expected Behavior

Report should be written in the host `out` directory

**Comment by seanmarpo** [comment]
Thanks for reporting this. I haven't tested the tool on Windows, so let me give it a try. I will report back what I find out.

**Comment by omnisnash** [comment]
Hi again,

I figured out the permission problem. When building the image on Windows, this message is displayed at the end:

````
SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.
````
The Dockerfile create the "app" user for the execution, which has no permission to create the "out" directory. So I added the write permission to the "app" user and now the the `PermissionError` is gone :)

However, the `out;C` is still generated on the host, and it doesn't contain the report :/

FYI, this is what the pwd command generate in my Git Bash terminal:
![image](https://user-images.githubusercontent.com/36193051/100355314-d4475300-2ff1-11eb-9431-df7502eb5e51.png)

**Comment by ajayep** [comment]
Hi,
Another way is to use _docker cp command_ on Windows
`docker cp a4ceae0d136f:/app/out/ D:\Docker`

**Comment by seanmarpo** [comment]
Hi there, I was able to test this on Windows. My suspicion is that that `bash` syntax isn't translating well into `cmd` or `powershell` depending on the Windows-based terminal you're using.

I was able to get the tool to output correctly by hand-typing out the path as such:
`docker run -v Z:\test\connect-security-req-tester\out:/app/out connect-sec-test https://elements.valiantys.com/elements/atlassian-connect.json`

Based on the number of warnings I received when setting this up, I can't recommend using the tool on Windows for now. But at least in my testing, it did complete without error.
